specs/core/1_custody-game.md

```diff
@@ -274,22 +293,22 @@ def process_custody_reveal(state: BeaconState,
                            reveal: CustodyKeyReveal) -> None:
     assert verify_custody_key(state, reveal)
     revealer = state.validator_registry[reveal.revealer_index]
-    current_custody_period = epoch_to_custody_period(get_current_epoch(state))
+    validators_next_custody_reveal_period = validator.latest_custody_reveal_period + 1
 
     # Case 1: non-masked non-punitive non-early reveal
     if reveal.mask == ZERO_HASH:
-        assert reveal.period == epoch_to_custody_period(revealer.activation_epoch) + revealer.custody_reveal_index
-        # Revealer is active or exited
-        assert is_active_validator(revealer, get_current_epoch(state)) or revealer.exit_epoch > get_current_epoch(state)
-        revealer.custody_reveal_index += 1
-        revealer.max_reveal_lateness = max(revealer.max_reveal_lateness, current_custody_period - reveal.period)
+        assert reveal.period == validators_next_custody_reveal_period
+        # Revealer is active or exited, but not withdrawn
+        assert is_slashable_validator(revealer, get_current_epoch(state))
+        revealer.latest_custody_reveal_period += 1
+        # Reward block proposer
         proposer_index = get_beacon_proposer_index(state, state.slot)
         increase_balance(state, proposer_index, base_reward(state, index) // MINOR_REWARD_QUOTIENT)
 
     # Case 2: masked punitive early reveal
     else:
-        assert reveal.period > current_custody_period
-        assert revealer.slashed is False
+        assert reveal.period >= validators_next_custody_reveal_period
```

Comment:
```suggestion
        assert reveal.period >= get_validators_current_custody_reveal_period(state, reveal.masker_index)
```
Only allow slashing for secrets that are still in use or will be used in the future